src/lib.rs

```diff
@@ -148,6 +148,14 @@ impl Client {
         Ok(token_response.token)
     }
 
+    //////////////// Accounts ////////////////
+
+    /// Log in with the given credentials
+    pub async fn login(&mut self, username: &str, password: &str) -> Result<()> {
```

Comment:
I wonder if we should have distinct `Client` and `UnauthenticatedClient` types so that you never risk using an unauthenticated client to call a method that requires auth.  (So `login` takes an unauthenticated client and returns an authenticated one, rather than mutating the client at hand.)

Reply:
Though in this case it would still be good to have a method that jumps you straight to the authenticated client rather than having to write `UAClient::new(...).login(...)`_.

Reply:
That's a good idea. We could definitely implement two structures that you can use `UnAuthenticatedClient::from(AuthenticatedClient::new(url, username, password)).register(newusername, newpassword)`.

I like that approach a lot, actually. Good thinking.